pandas/core/arrays/timedeltas.py

```diff
@@ -162,15 +162,29 @@ def _simple_new(cls, values, freq=None, dtype=_TD_DTYPE):
         result._freq = freq
         return result
 
-    def __new__(cls, values, freq=None, dtype=_TD_DTYPE):
+    def __new__(cls, values, freq=None, dtype=_TD_DTYPE, copy=False):
 
         freq, freq_infer = dtl.maybe_infer_freq(freq)
 
-        values = np.array(values, copy=False)
-        if values.dtype == np.object_:
-            values = array_to_timedelta64(values)
+        values, inferred_freq = sequence_to_td64ns(
+            values, copy=copy, unit=None)
+        if inferred_freq is not None:
+            if freq is not None and freq != inferred_freq:
+                raise ValueError('Inferred frequency {inferred} from passed '
```

Comment:
This is not covered according to codecov

Reply:
Thanks.  I'm fairly sure the matching line in `TimedeltaIndex.__new__` is covered by a targeted test.  I'll update that test to cover this too (and after #23675 I plan to do a pass to de-duplicate a bunch of the DTA/TDA/DTI/TDI constructors)